DemocracyClub/EveryElection
# Add migration for candidate death

The last of the cancellation reason implementation 🎉 

[Asana](https://app.asana.com/0/1204880927741389/1205170296632726)

The migration adds one of the four cancellation reasons to the Every Election database for each historic cancelled election that fits into the four categories:
1. `UNDER_CONTESTED` - fewer candidates than seats
2. `EQUAL_CANDIDATES` - equal number of candidates and seats
3. `NO_CANDIDATES` - Ronseal
4. `CANDIDATE_DEATH` - Also Ronseal. _This one is determined via metadata._

Elections which were cancelled/postponed for any other reason will not have `cancellation_reason` set at this time. This includes all of the Covid-19 cancellations as well as 2020 postponements for new unitary councils.  

The CSV was generated from data held in WCIVF on 2023-10-31 and this process should not need to be repeated as going forwards elections will have their reasons set at the same time as they are cancelled. 

To test:
- Make sure you're up to date on everything, including having a reasonably fresh database backup of EE
- Run the migrations
- The following SQL should yield a count of 389: `select count(*) from elections_election where cancellation_reason is not null;` (n.b. if you've been testing other cancellation reason stuff you may have a few more than 389, as I do)
- Any elections with metadata concerning candidate death should have a cancellation reason of `CANDIDATE_DEATH`



```[tasklist]
### PR Checklist
```
